x-pack/plugins/observability_solution/investigate/public/esql_widget/types.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IconType } from '@elastic/eui';
+import type { Ast } from '@kbn/interpreter';
+import type { InvestigateWidgetCreate } from '../../common';
+import type { GlobalWidgetParameters } from '../../common/types';
+
+// copied over from the Lens plugin to prevent dependency hell
+type TableChangeType = 'initial' | 'unchanged' | 'reduced' | 'extended' | 'reorder' | 'layers';
+
+interface Suggestion<T = unknown, V = unknown> {
+  visualizationId: string;
+  datasourceState?: V;
+  datasourceId?: string;
```

Comment:
nit: should source be capitalized?

```suggestion
  dataSourceState?: V;
  dataSourceId?: string;
```

Reply:
just copy pasted a type from the Lens plugin, it's not ours!